packages/core/src/particle/ParticleGenerator.ts

```diff
@@ -764,4 +805,201 @@ export class ParticleGenerator {
     out.push(vertexBufferBinding);
     return index;
   }
+
+  private _updateBoundingBox() {
+    const { min, max } = this._renderer._bounds;
+
+    min.set(Infinity, Infinity, Infinity);
+    max.set(-Infinity, -Infinity, -Infinity);
+
+    if (this._firstActiveElement < this._firstFreeElement) {
+      for (let i = this._firstActiveElement; i < this._firstFreeElement; i++) {
+        this._mergeParticleBoundsIntoRendererBounds(i);
+      }
+    } else {
+      for (let i = this._firstActiveElement; i < this._currentParticleCount; i++) {
+        this._mergeParticleBoundsIntoRendererBounds(i);
+      }
+      if (this._firstFreeElement > 0) {
+        for (let i = 0; i < this._firstFreeElement; i++) {
+          this._mergeParticleBoundsIntoRendererBounds(i);
+        }
+      }
+    }
+
+    if (this.main.simulationSpace === ParticleSimulationSpace.Local) {
+      const worldPosition = this._renderer.entity.transform.worldPosition;
+      min.add(worldPosition);
+      max.add(worldPosition);
+    }
+
+    this._addGravityModifierImpact();
+  }
```

Comment:
You should override `_updateBounds` in `ParticleRenderer` and combined with subsequent steps

1.  update frame world bounds list(only simulation space is world)
2. override `_updateBounds` with logic: Compute bounds with transform (if simulation space is local), Merge frame world bounds (if simulation space is world)